### ISSUE TITLE
@yieldparam does not work when passing the block to a method. 

### ISSUE THREAD

**Issue by Leguu** [castwide/solargraph#148]
Context: I'm creating a bot for Discord using a Ruby API.
When using a the API, you can create event handlers such as:
```ruby
Bot.message do |event|
  next 'Pong!' if event.content == 'Ping!'
end
```
`event` should be a `MessageEvent`, but solargraph doesn't recognise it as such. I've tested it with both VSCode and Atom. Doing `# @param event [Discordrb::Events::MessageEvent]` works, but having to retype that for every single message is inefficient. The relevant code from [Discordrb](https://github.com/meew0/discordrb): 
```ruby
...
# @yieldparam event [MessageEvent] The event that was raised.
# @return [MessageEventHandler] the event handler that was registered.
def message(attributes = {}, &block)
  register_event(MessageEvent, attributes, block)
end
...
```
Direct link to the file (Lines ~36): https://github.com/meew0/discordrb/blob/master/lib/discordrb/container.rb

Any help would be appreciated!

**Comment by castwide** [comment]
Confirmed. I cloned the discordrb repo and ran some tests. It works correctly if the `@yieldparam` tag type is fully qualified, e.g.:

```
# @yieldparam event [Discordrb::Events::MessageEvent] The event that was raised.
```

Since the `Events` module is included in `EventContainer`, the relative name `[MessageEvent]` should work too. I think I can have it fixed for the next patch release.

**Comment by castwide** [comment]
Fix published in gem v0.31.2.

**Comment by Leguu** [comment]
Thanks! It works now.
